### ISSUE TITLE
DatePicker fails accessibility testing. 

### ISSUE THREAD

**Issue by mhemesath** [cerner/terra-framework#469]
# Issue Description
Date picker fails accessibility testing

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [x] Bug
- [ ] Other

## Expected Behavior
Passes accessibility tests

## Current Behavior
* Control button fails from not having text. Related to #1058 
* Contrast ratio of selected day is not high enough. I think this is because the text is so small.
* The date picker doesn't generate a label for its input. *Note: this may just be an example update, but looking at doc I don't see anything on how to create a datepicker with a label*

## Steps to Reproduce

1. Install [Axe Chrome Extension](https://chrome.google.com/webstore/detail/axe/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US)
2. Navigate to a [date picker example](http://engineering.cerner.com/terra-core/#/tests/date-picker-tests/default) and open the datepicker by activating the control.
3. Run axe by going to developer tools and clicking axe tab.

## Environment

* Component Version:
* Browser Name and Version:
* Operating System and version (desktop or mobile):

**Comment by ryanthemanuel** [comment]
Closing this in favor of https://github.com/cerner/terra-framework/issues/464 which has more information on the exact issues.
